support-lib/jni/djinni_support.hpp

```diff
@@ -546,13 +546,45 @@ class JniEnum {
 
 protected:
     JniEnum(const std::string & name);
+    jclass enumClass() const { return m_clazz.get(); }
 
 private:
     const GlobalRef<jclass> m_clazz;
     const jmethodID m_staticmethValues;
     const jmethodID m_methOrdinal;
 };
 
+class JniFlags : private JniEnum {
```

Comment:
Private inheritance is a reasonable compromise on the prior discussion here.  Should keep any external uses of this class from having any confusion.